src/test/java/fr/nelson/you_are_the_hero/service/UserServiceTest.java

```diff
@@ -60,5 +64,52 @@ public void test_createNewUserWithExistingOne_ThrowsException() {
         Assertions.assertThrows(Exception.class, () -> this.userService.saveUser(authRequestDto));
     }
 
-}
+    @Test
+    public void testLoadUserByUsername_UserExists() {
+        AppUser appUser = new AppUser();
+        appUser.setUsername("jean neige");
+        appUser.setPassword("Ygrid");
+        appUser.setRole(Role.PLAYER);
+
+        when(appUserRepository.findByUsername("jean neige")).thenReturn(appUser);
+
+        UserDetails userDetails = userService.loadUserByUsername("jean neige");
+
+        assertNotNull(userDetails);
+        assertEquals("jean neige", userDetails.getUsername());
+        assertEquals("Ygrid", userDetails.getPassword());
+    }
 
+    @Test
+    public void testLoadUserByUsername_UserNotFound() {
+        when(appUserRepository.findByUsername("unknown-username")).thenReturn(null);
+
+        assertThrows(UsernameNotFoundException.class, () -> {
+            userService.loadUserByUsername("unknown-username");
+        });
+    }
+
+    @Test
+    public void testFindAppUserByUsername_UserExists() {
+        AppUser appUser = new AppUser();
+        appUser.setUsername("jean neige");
+        appUser.setPassword("Ygrid");
+        appUser.setRole(Role.PLAYER);
+
+        when(appUserRepository.findByUsername("jean neige")).thenReturn(appUser);
+
+        AppUser user = userService.findAppUserByUsername("jean neige");
+
+        assertNotNull(user);
+        assertEquals("jean neige", user.getUsername());
+    }
+
+    @Test
+    public void testFindAppUserByUsername_UserNotFound() {
+        when(appUserRepository.findByUsername("unknown-username")).thenReturn(null);
+
+        AppUser foundUser = userService.findAppUserByUsername("unknown-username");
```

Comment:
why not throw an exception if user not exist as for loadUserByUsername ?

Reply:
You're absolutely right. In the original context where the method was designed, there was no expectation that the user wouldn't be found, so the test was written based on that logic. However, I'll go ahead and create a new issue to address this.